cni/pkg/taint/taintcontroller.go

```diff
@@ -0,0 +1,414 @@
+package taint
+
+import (
+	"fmt"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	client "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/kubectl/pkg/util/podutils"
+
+	"istio.io/pkg/log"
+)
+
+type Controller struct {
+	clientset       client.Interface
+	podWorkQueue    workqueue.RateLimitingInterface
+	nodeWorkQueue   workqueue.RateLimitingInterface
+	podController   []cache.Controller
+	cachedPodsStore map[string]map[string]cache.Store // store sync with list watch given namespace and labelselector
```

Comment:
nit: if we have multiple keys (or even just one) its best to comment what the keys are

Reply:
add comments on the key information